cloudinary-core/src/test/java/com/cloudinary/api/signing/NotificationRequestSignatureVerifierTest.java

```diff
@@ -0,0 +1,71 @@
+package com.cloudinary.api.signing;
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class NotificationRequestSignatureVerifierTest {
+    @Test
+    public void testVerifySignature() {
+        NotificationRequestSignatureVerifier impl = new NotificationRequestSignatureVerifier("someApiSecret");
+
+        boolean actual = impl.verifySignature(
+                "{}",
+                "0",
+                "f9aa4471d2a88ff244424cca2444edf7d7ac3596");
+
+        assertTrue(actual);
+    }
+
+    @Test
+    public void testVerifySignatureFailWhenSignatureDoesntMatch() {
+        NotificationRequestSignatureVerifier impl = new NotificationRequestSignatureVerifier("someApiSecret");
```

Comment:
Why `impl`? should be called something more specific (e.g. `verifier`), there's no abstract/interface vs impl involved.